CONTRIBUTING.md

```diff
@@ -1,35 +1,88 @@
+
+1. Fork and clone the caiman repository    
+Go to the [https://github.com/flatironinstitute/CaImAn](Caiman repo) and hit the `Fork` button at the top right of the page. You now have Caiman on your own GitHub repo. On your computer, in your conda prompt, go to a directory where you want Caiman to download, and clone your personal Caiman repo: `git clone https://github.com/<your-username>/CaImAn.git` where <your-username> is replaced by your github username.
+2. Install in editable mode   
+In the previous step, you cloned your personal fork of Caiman. Detailed instructions for installing in development mode can be found on our [installation page](./docs/source/Installation.rst) for both [Windows](https://github.com/flatironinstitute/CaImAn/blob/master/docs/source/Installation.rst#installation-on-windows) and [Mac/Linux](https://github.com/flatironinstitute/CaImAn/blob/master/docs/source/Installation.rst#installation-on-macos-and-linux). While there are some differences, they both involve creating a conda environment from the `environment.yml` file you downloaded when you cloned the repo, activating that environment, and end with an "in place" install:
+
+        pip install -e . 
+
+    This installs Caiman directly from source code, leaving all the files you downloaded intact, and the `-e` stands for editable. When you edit the files, the changes will appear in your virtual environment.
+
+Note this section is partly based on the excellent [docs from Matplotlib](https://matplotlib.org/devdocs/devel/development_setup.html#installing-for-devs).
+
+
+## Second, work on a feature 
+Once you have a development environment, you can start working on the changes you envision. The main thing to remember is to follow the workflow in the diagram above. Let's say you want to work on feature `my_feature`. The first thing to do (label 1 in the diagram) is to create a new branch from the `dev` branch. Within the repo folder:
+
+    git checkout -b my_feature dev
+
+Then from within this branch you can do your usual work on the feature: work on stuff, add, commits, etc. This corresponds to the workflow in the blue boxes in the above diagram (label 2 in the figure).
+
+If you changed anything in the `.py` modules, we request that you run tests on your code before making a PR. You can test the code by typing `caimanmanager.py test`. 
+
+## Third, make a PR
+Once your feature branch is ready, it's time to make a PR (label 3 in the figure). This is fairly simple:
+
+    git push -u origin my_feature
+
+Then, go to your local fork of Caiman at your github repo, and a large green button will show at the top, giving you the option of making a `Pull request`. When you click it, this will send over your feature for review (be sure to push to merge with dev branch, not main, and to fill out the PR template that is auto-generated by Caiman with the information requested). 
```

Comment:
"be sure to push to merge with dev branch" -> "be sure to generate your PR targeting the dev branch"

(users won't be doing merging)